### ISSUE TITLE
[MappingApplication] Is there a missing synchronize in MPI?

### ISSUE THREAD

**Issue by marcnunezc** [KratosMultiphysics/Kratos#7780]
**Description**
Regarding #7728, I have been trying to do the following.

All run in MPI:
- Run a FluidDyn simulation (initial_model_part)
- Remesh with ParMmg (getting a main_model_part)
- Map the velocity and pressure fields from initial model_part to the main_model_part
- Run the FluidDyn simulation with the new mesh and fields (main_model_part).

What I am experiencing is that after mapping the fields, the new simulation genereates weird features in the field. This are caused due to a lack of synchornization of the values of the velocity across ranks. 

This is the correct mapped field.
![image](https://user-images.githubusercontent.com/32136457/98677003-ad83ee00-235c-11eb-95fa-98481f68dfac.png)

Field after some seconds of simulation (see the peaks on the velocity field).
![image](https://user-images.githubusercontent.com/32136457/98676965-9c3ae180-235c-11eb-8755-75760f54244b.png)

If after performing the mapping I run the lines: 
```python
main_model_part.GetCommunicator().SynchronizeVariable(KratosMultiphysics.VELOCITY)
main_model_part.GetCommunicator().SynchronizeVariable(KratosMultiphysics.PRESSURE)

```

The weird effects do not happen. 

![image](https://user-images.githubusercontent.com/32136457/98688258-efb42c00-236a-11eb-861a-ba4e9ce53d69.png)

If the velocity is compared among ranks, it can be seen that without the synchronization, different ranks have different values of VELOCITY on the same node. 

The mapping parameters that I use are:

```python
mapper_settings = KratosMultiphysics.Parameters("""{
"mapper_type": "nearest_element",
"interface_submodel_part_origin": "Parts_Fluid",
"interface_submodel_part_destination": "Parts_Fluid",
"search_radius" : 0.01,
"echo_level" : 0
}""")

# creating a mapper for distributed memory
mpi_mapper = KratosMapping.MapperFactory.CreateMPIMapper(
    initial_main_model_part,
    main_model_part,
    mapper_settings)
mpi_mapper.Map(KratosMultiphysics.VELOCITY, KratosMultiphysics.VELOCITY)
mpi_mapper.Map(KratosMultiphysics.PRESSURE, KratosMultiphysics.PRESSURE)


```

@philbucher Have you experienced this? Is there anything am I doing wrong? Is it expected that the user synchronizes the variables after mapping?

**Scope**
- Mapping application
- Fluid Dynamics application

**To Reproduce**
Find the script in this zip (run with 4 procs on master branch).

[mapping_sync_cylinder_test.zip](https://github.com/KratosMultiphysics/Kratos/files/5518144/mapping_sync_cylinder_test.zip)

Lines 126 and 127 are commented (which perform the sync). **MMG/ParMMG is not required**

**Expected behavior**
That variables are synchronized after mapping. 

**Environment**
Please specify your operating system, kratos branch, python version or any other library versions that you consider relevant.
E.g.:
- OS: Ubuntu 18.04
- Branch: master
- Python 3.6.9

**Comment by philbucher** [comment]
Hi @marcnunezc 
Yes this is known and has been discussed before with @roigcarlo and @RiccardoRossi. We were not sure if the sync should be done in the Mapper or not.
Since it can cause confusion I can also add it, I don't mind

any objections @roigcarlo @RiccardoRossi ?

**Comment by roigcarlo** [comment]
Not from my side, also ping @ddiezrod . They use it in their code so if you add it they must remove it.

**Comment by RiccardoRossi** [comment]
I didn't remember the discussion...please go ahead for me

Riccardo

**Comment by philbucher** [comment]
Ok seems like we are in agreement.
I will add the sync in the mapper then, I also don#t remember why we didn't do it in the first place...

I will leave this issue open until I fixed it

**Comment by rubenzorrilla** [comment]
Just mentioning that I'd also do the sync in the map. I think that the user will assume this is done in the vast majority of cases.

**Comment by riccardotosi** [comment]
Hello. Is this still open? I ask because we had an issue while running in MPI and using the MPI mapper that we solved by synchronizing manually the variables after the MPI mapping.

**Comment by philbucher** [comment]
sorry I missed/forgot this :sweat_smile: 
I will try to do it asap

**Comment by philbucher** [comment]
fixed in #9145
